`stdarch-test`: various cleanups

folkertdev opened the pull request:
My goal here was to make the instruction matching stricter, to prevent accepting e.g. `fminnm` when the test specifies the `fmin` instruction must be used. 

But especially on x86 there are lots of cases where a prefix is (deliberately?) used.

Anyway, I also made some other cleanups. Can be reviewed commit-by-commit.

Amanieu commented on crates/stdarch-test/src/lib.rs:
```diff
@@ -84,12 +81,26 @@ pub fn assert(shim_addr: usize, fnname: &str, expected: &str) {
+    let found = expected == "nop"
+        || instrs.iter().any(|instruction| {
+            instruction.starts_with(expected)
+            // Check that the next character is non-alphabetic. This prevents false negatives
```
I would expect digits (`is_ascii_digit`) to also be part of the instruction name.

Amanieu replied:
I don't see any intrinsics that specifically match on "ld". All the load instrinsics explicitly specify the number since this is an important part of the instruction.

Amanieu replied:
I would prefer if this included digits and had a way to explicitly opt-in to wildcard matching.

Amanieu commented on crates/stdarch-test/src/lib.rs:
```diff
@@ -84,12 +81,26 @@ pub fn assert(shim_addr: usize, fnname: &str, expected: &str) {
+            // Check that the next character is non-alphabetic. This prevents false negatives
+            // when e.g. `fminnm` was used but `fmin` was expected.
+            //
+            // TODO: resolve the conflicts (x86_64 has a bunch, probably others)
```
Are you intending to have this merged as it is, or wait for x86 conflicts to be resolved first? I'm happy to merge either way.

folkertdev commented:
I've updated the logic to use `is_ascii_alphanumeric`, though it's still commented out.

A selection of the false positives:

for x86 

```
---- core_arch::x86::avx::assert__mm256_store_pd_vmovap stdout ----
disassembly for stdarch_test_shim__mm256_store_pd_vmovap: 
	 0: vmovaps %ymm0,(%rdi)
	 1: vzeroupper
	 2: ret

thread 'core_arch::x86::avx::assert__mm256_store_pd_vmovap' panicked at crates/stdarch-test/src/lib.rs:204:9:
failed to find instruction `vmovap` in the disassembly

---- core_arch::x86::avx::assert__mm256_storeu_pd_vmovup stdout ----
disassembly for stdarch_test_shim__mm256_storeu_pd_vmovup: 
	 0: vmovups %ymm0,(%rdi)
	 1: vzeroupper
	 2: ret

thread 'core_arch::x86::avx::assert__mm256_storeu_pd_vmovup' panicked at crates/stdarch-test/src/lib.rs:204:9:
failed to find instruction `vmovup` in the disassembly

---- core_arch::x86::fma::assert__mm_fnmsub_ss_vfnmsub stdout ----
disassembly for stdarch_test_shim__mm_fnmsub_ss_vfnmsub: 
	 0: vfnmsub213ss %xmm2,%xmm1,%xmm0
	 1: ret

thread 'core_arch::x86::fma::assert__mm_fnmsub_ss_vfnmsub' panicked at crates/stdarch-test/src/lib.rs:204:9:
failed to find instruction `vfnmsub` in the disassembly

---- core_arch::x86::pclmulqdq::assert__mm_clmulepi64_si128_pclmul stdout ----
disassembly for stdarch_test_shim__mm_clmulepi64_si128_pclmul: 
	 0: pclmullqlqdq %xmm1,%xmm0
	 1: ret

thread 'core_arch::x86::pclmulqdq::assert__mm_clmulepi64_si128_pclmul' panicked at crates/stdarch-test/src/lib.rs:204:9:
failed to find instruction `pclmul` in the disassembly
```

for aarch64 I was thinking of `zip1`/`zip2`. I've now fixed that but there are a bunch that remain

```
---- core_arch::aarch64::neon::generated::assert_vmull_high_p64_pmull stdout ----
disassembly for stdarch_test_shim_vmull_high_p64_pmull: 
	 0: pmull2 v0.1q, v0.2d, v1.2d
	 1: mov x1, v0.d[1]
	 2: fmov x0, d0
	 3: ret

thread 'core_arch::aarch64::neon::generated::assert_vmull_high_p64_pmull' panicked at crates/stdarch-test/src/lib.rs:204:9:
failed to find instruction `pmull` in the disassembly

---- core_arch::aarch64::neon::generated::assert_vmull_high_p8_pmull stdout ----
disassembly for stdarch_test_shim_vmull_high_p8_pmull: 
	 0: pmull2 v0.8h, v0.16b, v1.16b
	 1: ret

thread 'core_arch::aarch64::neon::generated::assert_vmull_high_p8_pmull' panicked at crates/stdarch-test/src/lib.rs:204:9:
failed to find instruction `pmull` in the disassembly
```

it looks like for some instructions, the `2` version is used for some types/widths. So checking for the exact instruction would mean a bunch of duplication in the spec document.

Perhaps we can extend the macro to accept an explicit `prefix = "foo"` or similar, so that the default is to perform a strict compare, but we can opt-in into a less strict match.
